content/courses/Ada_For_The_C_Embedded_Developer/chapters/05_SPARK.rst

```diff
@@ -312,51 +488,62 @@ check requesting that :ada:`X` is supposed to be above 99. This corresponds to:
        Put_Line (Integer'Image (X));
     end Main;
 
+Now we also get the compiler warning about the pragma Assert condition. 
+When run, the failure of pragma Assert on line 7 raises the exception 
+Ada.Assertions.Assertion_Error. According to the expression in the 
+assertion, :ada:`X` is expected (incorrectly) to be above 99 after the 
+multiplication. (The exception name in the error message, 
+SYSTEM.ASSERTIONS.ASSERT_FAILURE, is a GNAT-specific alias for 
+Ada.Assertions.Assertion_Error.) 
 
-::
-
-    gprbuild -gnatp hello_world.adb
-    main
-    -49
-
-On this last example, we removed automatic check with ``-gnatp`` switch. The
-program will run without verification, and here provide a result that is
-inconsistent, -49, on data supposed to be positive. This corresponds to:
-
-.. code:: ada run_button project=Courses.Ada_For_C_Embedded_Dev.SPARK.Assert
-
-    with Ada.Text_IO; use Ada.Text_IO;
-
-    procedure Main is
-       X : Positive := 10;
-       pragma Suppress (All_Checks);
-    begin
-       X := X * 5;
-       X := X - 99;
-       Put_Line (Integer'Image (X));
-    end Main;
-
-It's interesting to see in the above example that code inspection could have
-pointed out that there would always be a problem. An alternative to code
-inspection is to use formal proof, which will find all places in the code that
-are not 100% safe. This can be done through the SPARK subset of the Ada
-language and the :program:`gnatprove` tool.
+It's interesting to see in the output that the compiler can detect some 
+violations at compile-time: 
 
-SPARK is a subset encompassing most of the Ada language, but for features that
-prevent the proof. As a disclaimer, this booklet is not aimed at providing a
-full introduction to the SPARK language and proof, but to present in a few
-examples what it is about.
+::
 
-As it turns out, our code is already SPARK compliant. We just need to enable
-SPARK checks with the :ada:`SPARK_Mode` aspect:
+    main.adb:7:19: warning: assertion will fail at run time
+    main.adb:7:21: warning: condition can only be True if invalid values present
+    main.adb:8:11: warning: value not in range of type "Standard.Positive"
+
+Generally speaking, a complete analysis of that sort is beyond the scope 
+of compilers and they will not find all errors prior to execution, even 
+ones we can detect ourselves by inspection. More errors can be found by 
```

Comment:
Question by a non-native speaker: shouldn't it be "even _the_ ones we can detect..."?